File: packages/parallax_sdk/src/jobs.ts
// Job types shared across parallax packages

export interface ParallaxJobData {
  action: string;
  media: string;
  model: string;
  script: string;
  args: string[];
  scriptBase: string;
  uvPath: string;
}

export interface ParallaxJobResult {
  outputPath: string;
}

export interface PythonProgress {
  step: string;
  pct: number;
  frame?: number;
  total?: number;
  output?: string;
  error?: string;
}

File: server/jobs.py
from __future__ import annotations

from pydantic import BaseModel

from server.schemas import JobResult as JobResult  # noqa: F401 — re-exported for consumers


class JobData(BaseModel):
    action: str
    media: str
    model: str
    cmd: list[str]


class PythonProgress(BaseModel):
    step: str
    pct: float
    frame: int | None = None
    total: int | None = None
    output: str | None = None
    error: str | None = None
